index.js

```diff
@@ -1,634 +1,465 @@
-var tcp = require('../../tcp')
-var instance_skel = require('../../instance_skel')
+const { InstanceBase, Regex, runEntrypoint, TCPHelper, combineRgb, InstanceStatus } = require('@companion-module/base')
 var presets = require('./presets')
-var debug
-var log
 
-function instance(system, id, config) {
-	var self = this
+class CueTimerInstance extends InstanceBase {
+	constructor(internal) {
+		super(internal)
 
-	// super-constructor
-	instance_skel.apply(this, arguments)
-
-	self.actions() // export actions
-
-	return self
-}
-instance.prototype.hexToRgb = function (hex) {
-	var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex)
-	return result
-		? {
-				r: parseInt(result[1], 16),
-				g: parseInt(result[2], 16),
-				b: parseInt(result[3], 16),
-		  }
-		: null
-}
-String.prototype.lpad = function (padString, length) {
-	var str = this
-	while (str.length < length) str = padString + str
-	return str
-}
-
-instance.prototype.updateConfig = function (config) {
-	var self = this
-	var resetConnection = false
-
-	if (self.config.host != config.host || self.config.port != config.port) {
-		resetConnection = true
+		String.prototype.lpad = function (padString, length) {
+			var str = this
+			while (str.length < length) str = padString + str
+			return str
+		}
 	}
 
-	self.config = config
-
-	if (resetConnection === true || self.socket === undefined) {
-		self.initTCP()
+	hexToRgb(hex) {
+		var result = /^#?([a-f\d]{2})([a-f\d]{2})([a-f\d]{2})$/i.exec(hex)
+		return result
+			? {
+					r: parseInt(result[1], 16),
+					g: parseInt(result[2], 16),
+					b: parseInt(result[3], 16),
+			  }
+			: null
 	}
 
-	self.setupEventListeners.bind(self)();
-}
-
-instance.prototype.init = function () {
-	var self = this
-
-	debug = self.debug
-	log = self.log
-
-	self.variable1Value = '';
-	self.variable2Value = '';
-	self.variable3Value = '';
-	self.variable4Value = '';
-	self.variable5Value = '';
-
-	self.initTCP()
-	self.setupEventListeners.bind(this)();
-	self.feedbacks()
-	self.presets()
-	self.variables()
-
-	self.setVariable('hours', '')
-	self.setVariable('minutes', '')
-	self.setVariable('seconds', '')
-	self.setVariable('name', '')
-	self.setVariable('speed', '')
-	self.setVariable('endTime', '')
-	self.setVariable('nextTimerName', '')
-	self.setVariable('nextTimerDuration', '')
-	self.bgColor = self.rgb(0, 0, 0)
-	self.fgColor = self.rgb(255, 255, 255)
-	self.buttonStates = {
-		Fullscreen: false,
-		NDI: false,
-		Message: false,
-		STM: false,
-		Clock: false,
-		Pause: false,
-		Blackout: false,
-	}
-	self.timers = {}
-}
+	updateConfig(config) {
```

Comment:
This should be `async configUpdated`

Reply:
Done, but I don't remember seeing it in [Upgrading a module built for Companion 2.x ](https://github.com/bitfocus/companion-module-base/wiki/Upgrading-a-module-built-for-Companion-2.x) maybe I missed it